ksql-engine/src/main/java/io/confluent/ksql/function/udf/json/JsonExtractStringKudf.java

```diff
@@ -16,62 +16,74 @@
 
 package io.confluent.ksql.function.udf.json;
 
+import com.google.common.collect.ImmutableList;
+
 import com.fasterxml.jackson.databind.JsonNode;
 import com.fasterxml.jackson.databind.ObjectMapper;
-import com.google.common.collect.ImmutableList;
+import com.fasterxml.jackson.databind.node.ArrayNode;
+
+import java.io.IOException;
+import java.util.List;
+
 import io.confluent.ksql.function.KsqlFunctionException;
 import io.confluent.ksql.function.udf.Kudf;
-import io.confluent.ksql.util.KsqlException;
 import io.confluent.ksql.util.json.JsonPathTokenizer;
 
-import java.io.IOException;
-
 public class JsonExtractStringKudf implements Kudf {
   private static final ObjectMapper OBJECT_MAPPER = new ObjectMapper();
```

Comment:
this `ObjectMapper` is effectively shared across all threads of all queries using this UDF. If there is any synchronization in it then this is potentially going to have an impact across all of those queries. I think it would be better to remove `static`

Reply:
I hear your point, but a shared static instance is the recommended approach from the authors of the library, so we can be fairly confident that any performance issues, due to synchronization, would have been fixed.  

This type is thread safe _only_ if you don't change its settings once its shared.

So my view would be to leave it shared. But if you feel strongly the other way then I'm happy enough to change it.

Reply:
I'm ok with it as is for now. I found this: https://stackoverflow.com/questions/3907929/should-i-declare-jacksons-objectmapper-as-a-static-field one of the answers suggests that there is a performance penalty. That said - we probably shouldn't change it unless we tested it.

Reply:
Swapped to the `ObjectReader` instead.